proc/proc.go

```diff
@@ -359,10 +359,27 @@ func (dbp *Process) setChanRecvBreakpoints() (int, error) {
 
 // Resume process.
 func (dbp *Process) Continue() error {
+	// all threads stopped over a breakpoint are made to step over it
 	for _, thread := range dbp.Threads {
-		err := thread.Continue()
-		if err != nil {
-			return fmt.Errorf("could not continue thread %d %s", thread.Id, err)
+		if thread.CurrentBreakpoint != nil {
+			if err := thread.Step(); err != nil {
+				return err
+			}
+			if err := thread.SetCurrentBreakpoint(thread.CurrentBreakpoint); err != nil {
```

Comment:
I don't think moving this code in `dbp.run` is appropriate for `next` and `step`: `next` needs to set its temp breakpoints before continuing and `step` shouldn't call `resume` at all. Also moving this into `thread.Step` would break `step` (because it steps one instruction anyway, not just if the thread is on a breakpoint).